nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureProxyUtils.java

```diff
@@ -0,0 +1,115 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.azure.storage.utils;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.Collection;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.context.PropertyContext;
+import org.apache.nifi.proxy.ProxyConfiguration;
+import org.apache.nifi.proxy.ProxySpec;
+import org.apache.nifi.util.StringUtils;
+
+public class AzureProxyUtils {
+    private static final ProxySpec[] PROXY_SPECS = {ProxySpec.HTTP, ProxySpec.SOCKS};
+
+    private static ProxyOptions.Type getProxyOptionsTypeFromProxyType(final Proxy.Type proxyType) {
+        for (final ProxyOptions.Type item : ProxyOptions.Type.values()) {
+            if (item.toProxyType() == proxyType) {
+                return item;
+            }
+        }
+        return null;
+    }
+
+    public static final PropertyDescriptor PROXY_CONFIGURATION_SERVICE = ProxyConfiguration
+            .createProxyConfigPropertyDescriptor(false, PROXY_SPECS);
+
+    public static HttpClient createHttpClient(final PropertyContext propertyContext) {
+        final ProxyConfiguration proxyConfig = ProxyConfiguration.getConfiguration(propertyContext);
+        final ProxyOptions proxyOptions = getProxyOptions(proxyConfig);
+
+        final HttpClient client = new NettyAsyncHttpClientBuilder()
+            .proxy(proxyOptions)
+            .build();
+
+        return client;
+    }
+
+    public static void validateProxySpec(final ValidationContext context, final Collection<ValidationResult> results) {
+        final ProxyConfiguration proxyConfig = ProxyConfiguration.getConfiguration(context);
+
+        final String proxyServerHost = proxyConfig.getProxyServerHost();
+        final Integer proxyServerPort = proxyConfig.getProxyServerPort();
+        final String proxyServerUser = proxyConfig.getProxyUserName();
+        final String proxyServerPassword = proxyConfig.getProxyUserPassword();
+
+        if ((StringUtils.isNotBlank(proxyServerHost) && proxyServerPort == null)
```

Comment:
These checks are relevant in case of `HTTP` and `SOCKS` proxy but not for `DIRECT`.
In case of `DIRECT` (that is no proxy), it could be checked that no other properties are specified.